### ISSUE TITLE
Is it possible to use mobx-persist with mobx-react-lite ?

### ISSUE THREAD

**Issue by lucas-wilmart** [mobxjs/mobx-react-lite#253]
Hello,

Thanks for this amazing library I really enjoy it for now. I'm a beginner in MobX but I've been using React and TypeScript for quite a while now.

I'm currently working on a React/TypeScript/MobX project and the only problem I've got so far is that I don't get how to use the library "mobx-persist" with mobx-react-lite. For instance I don't understand how to use the persist function since I don't have a defined schema (I'm using TypeScript).

Do you have any tips or guidelines to achieve this ?

Otherwise do you recommend using other libraries ? Should I be using "mobx-react" instead ?

**Comment by mweststrate** [comment]
mobx-react-lite is a rendering library, like mobx-react. So it doesn't
relate to mobx-persist at all. It sounds like you should be asking your
question either at mobx-persist or mobx. Anyhow, make sure that when you
ask a question it is a little more specific, showing some actual problem
and code. "how do I use library X" is quite a broad question, which is very
hard to answer beyond 'read the readme'

**Comment by lucas-wilmart** [comment]
You're right my question was a bit blurry indeed. But my main concern was still legit since persisting data stores is a very common issue and mobx-persist is definitely the most popular one in the mobx community and I was only looking for advices since mobx-persist documentation looked a bit incompatible with mobx-persist and I use this lib without store classes. This question should have been probably posted to mobx-persist issues, though.

Anyway, I gave up on mobx-persist and successfully implemented mobx-sync with mobx-react-lite. Looks like it's working fine for now.

If anyone still have suggestions or tips about this matter I'm still interested though

**Comment by FredyC** [comment]
@lucas-wilmart There is some weird misconception that `mobx-react` (+ lite) is about providing a way to structure your state. While in reality, the major reason for those libraries to exists is **to have reactive components**. I suppose this is sort of drag from the past where Provider+inject had to be included in the library because it would be too annoying with legacy context otherwise. 

As long as you have MobX observable in a variable, you can use it with anything else in the ecosystem. Doesn't matter if it's constructed through class sugar or `useLocalStore` or another abstraction layer.

There are more robust solutions like mobx-keystone and mobx-state-tree which do have serialization mechanics baked-in so any sort of persistence is piece of cake.

**Comment by lucas-wilmart** [comment]
@FredyC Thanks for your answer, I'm beginning to realize my mistake and how I misused this library, thanks for pointing that out. It's probably because I was looking for a redux alternative and lots of articles / videos I read/watch brought to my attention MobX (and react-mobx-lite) as a replacement for redux. 

The part of the official documentation which describe how to implement a global store with MobX ( https://mobx-react.js.org/state-local#what-about-global-store ) is mainly why I must have got the point of this library all wrong. To be honest, it's a bit confusing to me. Probably because having used redux for more than 3 years now, I might have tried to hard to make it look like my redux ecosystem. I did remember seeing the mention that mobx-state-tree was more appropriate for complex structure stores but since I wanted to keep it simple, I didn't looked a lot into it.

I will definitely check these two libraries you mentioned. All of this is still a bit confusing to me, but at least I learned a lot about React Context and mobx and hope to get a great setup working for my web app.

Thanks a lot
